packages/toolkit/src/query/apiTypes.ts

```diff
@@ -1,30 +1,358 @@
 import type {
-  EndpointDefinitions,
+  ActionCreatorWithPayload,
+  Middleware,
+  Reducer,
+  ThunkAction,
+  ThunkDispatch,
+  UnknownAction,
+} from '@reduxjs/toolkit'
+import type { BaseQueryFn } from './baseQueryTypes'
+import type {
+  ApiEndpointMutation,
+  ApiEndpointQuery,
+  CombinedState,
+  CoreModule,
+  coreModuleName,
+  InternalActions,
+  MutationActionCreatorResult,
+  MutationKeys,
+  PatchQueryDataThunk,
+  PrefetchOptions,
+  QueryActionCreatorResult,
+  QueryKeys,
+  RootState,
+  SliceActions,
+  ThunkWithReturnValue,
+  UpdateQueryDataThunk,
+  UpsertQueryDataThunk,
+} from './core'
+import type { CreateApiOptions } from './createApi'
+import type {
   EndpointBuilder,
   EndpointDefinition,
+  EndpointDefinitions,
+  MutationDefinition,
+  QueryArgFrom,
+  QueryDefinition,
+  TagDescription,
   UpdateDefinitions,
 } from './endpointDefinitions'
 import type {
-  UnionToIntersection,
   NoInfer,
+  UnionToIntersection,
   WithRequiredProp,
 } from './tsHelpers'
-import type { CoreModule } from './core/module'
-import type { CreateApiOptions } from './createApi'
-import type { BaseQueryFn } from './baseQueryTypes'
-import type { CombinedState } from './core/apiState'
-import type { UnknownAction } from '@reduxjs/toolkit'
 
 export interface ApiModules<
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   BaseQuery extends BaseQueryFn,
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   Definitions extends EndpointDefinitions,
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   ReducerPath extends string,
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   TagTypes extends string,
-> {}
+> {
+  [coreModuleName]: {
```

Comment:
:question: I assume this was added because we're no longer doing interface merging, or something along those lines?

Reply:
We can still do `interface` merging, I only removed `declare module`s that were using relative paths. Since now we are bundling the type defs, and relative paths will no longer be applicable.